internal/backend/crypto/gpg/cli/encrypt.go

```diff
@@ -25,25 +28,37 @@ func (g *GPG) Encrypt(ctx context.Context, plaintext []byte, recipients []string
 		args = append(args, "--trust-model=always")
 	}
 
+	if len(recipients) == 0 {
+		debug.Log("recipients list is empty!")
+	} else {
+		debug.Log("recipients are %s", recipients)
+	}
+	var badRecipients []string
 	for _, r := range recipients {
 		kl, err := g.listKeys(ctx, "public", r)
 		if err != nil {
 			debug.Log("Failed to check key %s. Adding anyway. %s", err)
 		} else if len(kl.UseableKeys(gpg.IsAlwaysTrust(ctx))) < 1 {
-			out.Printf(ctx, "Not using invalid key %s for encryption. (Check its expiration date or its encryption capabilities.)", r)
+			badRecipients = append(badRecipients, r)
+			errmsg := fmt.Sprintf("Not using invalid key %s for encryption. Check its expiration date, its encryption capabilities and trust.", r)
+			debug.Log(errmsg)
+			out.Printf(ctx, errmsg)
```

Comment:
Is the `debug.Log` really needed? I thought `out.Printf` was logging already:
https://github.com/gopasspw/gopass/blob/d00a8e1dbba5d3a8b4db19ee54f1734fa184dde4/internal/out/print.go#L47-L54

Is `ctxutil.IsHidden` returning `true` in this `Encrypt` method?

Reply:
> Is `ctxutil.IsHidden` returning `true` in this `Encrypt` method?

Good question... I think it depends on the context of the function calling this `Encrypt` method.. Could it be that when someone runs `gopass` from withing the web extension it is hidden? And that's why I experienced lack of useful logs when I tried to debug why it hadn't worked for me?

Reply:
I think @AnomalRoil is right and we don't need the extra `debug.Log` statement, but also it doesn't hurt. So let's not block on this further.